src/test/java/com/puppycrawl/tools/checkstyle/utils/CommonUtilTest.java

```diff
@@ -489,6 +498,32 @@ public void testGetUriByFilenameFindsRelativeResourceOnClasspathPrefix() throws
         assertEquals("Checker", config.getName(), "Unexpected config name!");
     }
 
+    @Test
+    public void testLoadSuppressionsUriSyntaxException() throws Exception {
+        final URL configUrl = mock(URL.class);
+        when(configUrl.toURI()).thenThrow(URISyntaxException.class);
+        try (MockedStatic<CommonUtil> utilities =
+                     mockStatic(CommonUtil.class, CALLS_REAL_METHODS)) {
+            final String fileName = "/suppressions_none.xml";
+            utilities.when(() -> CommonUtil.getCheckstyleResource(fileName)).thenReturn(configUrl);
+
+            try {
+                CommonUtil.getUriByFilename(fileName);
+                fail("Exception is expected");
+            }
+            catch (CheckstyleException ex) {
```

Comment:
why dont use assertThrows? it is more convinient way than try/catch with fail inside

Reply:
done